src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -63,6 +67,10 @@ const useFetchChannelHook = () => {
       channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
       channel.myUserId = signedProfileId;
     }
+
+    const isDeletedMessage = channel.firstMessage?.message_type === MESSAGE_TYPE_DELETED;
+    if (isDeletedMessage) channel.firstMessage.text = DELETED_MESSAGE_TEXT;
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is modifying the `channel.firstMessage.text` directly if it's a deleted message. This could potentially lead to issues if other parts of the code rely on the original text of the first message. Consider creating a new property to hold the display text instead.
```diff
-    const isDeletedMessage = channel.firstMessage?.message_type === MESSAGE_TYPE_DELETED;
-    if (isDeletedMessage) channel.firstMessage.text = DELETED_MESSAGE_TEXT;
+    const isDeletedMessage = channel.firstMessage?.message_type === MESSAGE_TYPE_DELETED;
+    if (isDeletedMessage) channel.firstMessage.displayText = DELETED_MESSAGE_TEXT;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->